common/src/java/com/zimbra/common/util/DateTimeUtil.java

```diff
@@ -41,4 +43,44 @@ public static boolean sameDay(Date t1, Date t2, TimeZone tz) {
             cal1.get(Calendar.MONTH) == cal2.get(Calendar.MONTH) &&
             cal1.get(Calendar.DAY_OF_MONTH) == cal2.get(Calendar.DAY_OF_MONTH);
     }
+
+    public static boolean checkWithinTime(Timestamp timestamp, int timegap, TimeUnit timeunit) {
```

Comment:
Can you simplify this method using `TimeUnit#valueOf(timeunit.name())` and `Timestamp#before`?

Reply:
Additionally this is an easy enough method to add a couple units tests to cover.

Reply:
done